### PR TITLE
Improving testing uplift data generation

### PR DESCRIPTION
**Pull request by lawinslow**
Current implementation of data generation `dgp` does not include any interaction between features and treatment. I added `sim_pte` as translated from the R `uplift` package which supports that functionality (though it too could be improved).

- Added sim_pte translated from R uplift package
- Aligned (somewhat) with exisitng function paramter options
- Changed default seed to be random for both functions
- Version increment because we really aren't on 0.0.1 anymore. 

(I do not know if I did documentation correctly)

### CODE REVIEW

**Comment by rsyi** [comment]
Luke this is awesome! I don't have access to merge PRs in the repo anymore, but you should ask Ben Clark to get you master access (or ask Will frost to merge). 

The problem with the documentation is that it's tied to my repo (rsyi/pylift), because I couldn't sign up for readthedocs with the wayfair organization email. But I'd be happy to transfer the docs to you guys if you need/figure out how to do it. In the meantime, I can merge the wayfair repo into mine once your PR goes through and it should update.

File: pylift/generate_data.py [hunk 1]
```diff
@@ -63,3 +67,93 @@ def dgp(N=1000, n_features=3, beta=[-3,-8,13,-8], error_std=0.5, tau=3, tau_std=
 
     df = pd.DataFrame(np.concatenate((X, treatment.reshape(-1,1), y.reshape(-1,1)), axis=1), columns=names)
     return df
+
+
+
+def sim_pte(N = 1000, n_features = 20, beta = None, rho = 0, sigma =  np.sqrt(2), beta_den = 4, discrete_outcome=False, seed=None):
```
Comment (rsyi) [inline pylift/generate_data.py]: for pep8 style, you should remove the spaces around the equal signs in function definitions. but not a big deal - we can autopep8 it later.

**Review by rsyi — APPROVED** [review]

File: pylift/generate_data.py (hunk 1 above)
Comment (lawinslow) [reply pylift/generate_data.py]: I'll push a fix. This is what PR and review is for.

Comment (lawinslow) [reply pylift/generate_data.py]: Fix pushed.

**Pull request merged**